airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/S3ConsumerFactory.kt

```diff
@@ -68,19 +68,20 @@ class S3ConsumerFactory {
                 val stream = writeConfig.streamName
                 val outputBucketPath = writeConfig.outputBucketPath
                 val pathFormat = writeConfig.pathFormat
-                if (mustCleanUpExistingObjects(writeConfig, storageOperations)) {
+                if (!isAppendSync(writeConfig)) {
                     LOGGER.info {
-                        "Clearing storage area in destination started for namespace $namespace " +
+                        "Listing objects to cleanup for namespace $namespace " +
                             "stream $stream bucketObject $outputBucketPath pathFormat $pathFormat"
                     }
-                    storageOperations.cleanUpBucketObject(
-                        namespace,
-                        stream,
-                        outputBucketPath,
-                        pathFormat
+                    writeConfig.objectsFromOldGeneration.addAll(
```

Comment:
Another raised CDK question: what's the best way to handle accumulated stream state. Is appending to the config the right way? Does it make things less manageable?

Reply:
this is only remnant in S3 land, we avoided attaching to WriteConfig (or rather killed that object all together) and saving states at instance level of that `StreamOperation` class. If there is a "state" machine then the FSM executor should accumulate and pass it along right ?